cfgmgr/sflowmgr.cpp

```diff
@@ -194,23 +194,32 @@ void SflowMgr::sflowGetPortInfo(vector<FieldValueTuple> &fvs, SflowPortInfo &loc
     fvs.push_back(fv2);
 }
 
-void SflowMgr::sflowCheckAndFillValues(string alias, vector<FieldValueTuple> &fvs)
+void SflowMgr::sflowCheckAndFillValues(string alias, vector<FieldValueTuple> &values,
+                                       vector<FieldValueTuple> &fvs)
 {
     string rate;
     bool admin_present = false;
     bool rate_present = false;
 
-    for (auto i : fvs)
+    for (auto i : values)
     {
         if (fvField(i) == "sample_rate")
         {
             rate_present = true;
             m_sflowPortConfMap[alias].rate = fvValue(i);
+            FieldValueTuple fv(fvField(i), fvValue(i));
+            fvs.push_back(fv);
         }
         if (fvField(i) == "admin_state")
         {
             admin_present = true;
             m_sflowPortConfMap[alias].admin = fvValue(i);
+            FieldValueTuple fv(fvField(i), fvValue(i));
+            fvs.push_back(fv);
+        }
+        if (fvField(i) == "NULL")
```

Comment:
When the ConfigDB has a key containing only one attribute and that attribute is being deleted from KLISH, REST or gNMI using the OC YANG or sonic YANG then a NULL/NULL field is inserted into the ConfigDB by the transformer module in management framework.

For example:
`sonic(conf-if-Ethernet0)# sflow sampling-rate 256
`
```
127.0.0.1:6379[4]> hgetall "SFLOW_SESSION|Ethernet0"
1) "sample_rate"
2) "256"

```

`sonic(conf-if-Ethernet0)# no sflow sampling-rate
`
```
127.0.0.1:6379[4]> hgetall "SFLOW_SESSION|Ethernet0"
1) "NULL"
2) "NULL"
```

NULL field is handled in sflowmgr to avoid NULL:NULL field value in APPLDB SFLOW_SESSION_TABLE:Ethernet0

Reply:
Got it. Thanks for the explanation.